File: SOC/src/cc/game/soc/core/IDistances.java
package cc.game.soc.core;

import java.util.List;

public interface IDistances {

	public final static byte DISTANCE_INFINITY = Byte.MAX_VALUE;
	
	List<Integer> getShortestPath(int fromVertex, int toVertex);
	
	int getDist(int from, int to);

}

File: SOC/src/cc/game/soc/core/IDistances.kt
package cc.game.soc.core

interface IDistances {
	fun getShortestPath(fromVertex: Int, toVertex: Int): List<Int>
	fun getDist(from: Int, to: Int): Int

	companion object {
		const val DISTANCE_INFINITY = Byte.MAX_VALUE
	}
}